From 71f3542672c0c2f7779b8b6b85e2c3a441c96ff5 Mon Sep 17 00:00:00 2001
From: "haotian.qu" <haotian.qu@amlogic.com>
Date: Mon, 24 Jun 2024 20:38:21 +0800
Subject: [PATCH 1/9] hdmionly: kirkstone support hdmionly [1/3]

PD#

Problem:
kirkstone support hdmionly.

Solution:
kirkstone support hdmionly.

Verify:
local

Signed-off-by: haotian.qu <haotian.qu@amlogic.com>
Change-Id: I64a118fff54e6d636cdffa3e7b9b86194dc17afe
---
 recipes-bsp/u-boot/files/hdmitx_only.cfg      |  2 +
 .../linux/aml_dtoverlay/hdmionly_overlay.dtsi | 44 ++++++++++++++++++-
 .../aml_dtoverlay/t7_debian_overlay.dtsi      | 14 ------
 3 files changed, 45 insertions(+), 15 deletions(-)

diff --git a/recipes-bsp/u-boot/files/hdmitx_only.cfg b/recipes-bsp/u-boot/files/hdmitx_only.cfg
index 816413d6..6110f4a2 100644
--- a/recipes-bsp/u-boot/files/hdmitx_only.cfg
+++ b/recipes-bsp/u-boot/files/hdmitx_only.cfg
@@ -1 +1,3 @@
 CONFIG_HDMITX_ONLY=y
+CONFIG_AML_DT_OVERLAY=y
+CONFIG_MULTI_DTB=y
diff --git a/recipes-kernel/linux/aml_dtoverlay/hdmionly_overlay.dtsi b/recipes-kernel/linux/aml_dtoverlay/hdmionly_overlay.dtsi
index 783a6430..8302364a 100644
--- a/recipes-kernel/linux/aml_dtoverlay/hdmionly_overlay.dtsi
+++ b/recipes-kernel/linux/aml_dtoverlay/hdmionly_overlay.dtsi
@@ -5,7 +5,49 @@
     fragment@0 {
         target-path = "/meson-amvideom";
         __overlay__ {
-            status = "disabled";
+            display_device_cnt = <1>;
+            vpp1_layer_count = <0>;
+        };
+    };
+    fragment@1 {
+        target-path = "/fb";
+        __overlay__ {
+            display_device_cnt = <1>;
+        };
+    };
+    fragment@2 {
+        target-path = "/lcd1";
+        __removenode__ {
+        };
+    };
+    fragment@3 {
+        target-path = "/lcd2";
+        __removenode__ {
+        };
+    };
+    fragment@4 {
+        target-path = "/lcd";
+        __removenode__ {
+        };
+    };
+    fragment@5 {
+        target-path = "/drm-vpu@0xff900000/port@1";
+        __overlay__ {
+            endpoint@2 {
+                remote-endpoint = <0>;
+            };
+            endpoint@3 {
+                remote-endpoint = <0>;
+            };
+            endpoint@4 {
+                remote-endpoint = <0>;
+            };
+        };
+    };
+    fragment@6 {
+        target-path = "/drm-subsystem";
+        __overlay__ {
+            crtc_masks = <7 7 1>;
         };
     };
 };
\ No newline at end of file
diff --git a/recipes-kernel/linux/aml_dtoverlay/t7_debian_overlay.dtsi b/recipes-kernel/linux/aml_dtoverlay/t7_debian_overlay.dtsi
index b60bbbe4..7a2cfc19 100644
--- a/recipes-kernel/linux/aml_dtoverlay/t7_debian_overlay.dtsi
+++ b/recipes-kernel/linux/aml_dtoverlay/t7_debian_overlay.dtsi
@@ -3,20 +3,6 @@
 
 / {
     fragment@0 {
-        target-path = "/lcd";
-        __overlay__ {
-            status = "disabled";
-        };
-    };
-
-    fragment@1 {
-        target-path = "/lcd2";
-        __overlay__ {
-            status = "disabled";
-        };
-    };
-
-    fragment@2 {
         target-path = "/aml_bt";
         __overlay__ {
             power_down_disable = <1>;
-- 
2.25.1

